### PR TITLE
fix: input share store safe v2

### PR DESCRIPTION
**Pull request by guru-web3**

### CODE REVIEW

File: packages/default/test/shared.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ export const sharedTestCases = (mode, torusSP, storageLayer) => {
 
       const tb2 = new ThresholdKey({ serviceProvider: customSP, storageLayer: customSL, manualSync: mode });
       await tb2.initialize({ neverInitializeNewKey: true });
-      tb2.inputShareStore(resp1.deviceShare);
+      await tb2.inputShareStoreSafe(resp1.deviceShare);
```
Comment (ieow) [inline packages/default/test/shared.js:91]: are we deprecating inputShareStore and use only inputShareStoreSafe for all usage ?
if so, we can check for if polynomial_id is equal with the tkey's latest polynomial and update it without catchupToLatestShare.

File: packages/core/src/core.ts [hunk 2]
```diff
@@ -815,6 +815,9 @@ class ThresholdKey implements ITKey {
     const latestShareRes = await this.catchupToLatestShare({ shareStore: ss, includeLocalMetadataTransitions: true });
     // if not in poly id list, metadata is probably outdated
     // is !this.metadata.polyIDList.includes(latestShareRes.latestShare.polynomialID)
+    if (!latestShareRes.shareMetadata.polyIDList.find((tuple) => tuple[0] === this.metadata.getLatestPublicPolynomial().getPolynomialID())) {
```
Comment (ieow) [inline packages/core/src/core.ts:818]: move this before the comment as the comment are meant for line 821

can add comment to check if the latest share is part of the current tkey instances for this new code

**Review by ieow — APPROVED** [review]
lgtm

File: packages/default/test/shared.js (hunk 1 above)
Comment (metallicalfa2) [reply packages/default/test/shared.js:91]: hmm, we should make inputShareStore internal or mark internal somehow. With existence of core kit, tkey has more room to be "swiss army knife"

**Pull request merged**